008-strimzi-ui.md

```diff
@@ -0,0 +1,119 @@
+
+In addition to the implementation detailed above, I would propose that any UI contributed to Strimzi be managed and developed as follows:
+
+- Be developed in a dedicated repository inside the Strimzi Github organisation
+- Both the server and client are contained in this repo - worked as sub modules in a monorepo
+- That design/discussion/implementation and defect issues are kept solely in this repository
+- All development is done in a behavioural manner - focusing on the end user, and the task they are trying to achieve (see [topic list](#topic-list-page) for an example)
+- That the repository operate in a cloud like CI/CD manner; lots of small/little and often deliveries of new function, appropriately sized and gated, allowing for the UI to be shipped at any time
+- That automation, such as [GitHub actions](https://github.com/features/actions) for example, is used as much as possible to support this CI/CD model - eg, automated dependency updates, automated testing run on PR, issue triage etc
+
+### Phased delivery
+
+A UI is more than just what a user sees. It also needs to be supported by backend data, services, and resources. To this end, and in bearing in mind the cloud development model mentioned above, all UI capability (both front and backend) should be provided in a phased manner. [I would suggest initially that a topic listing page is the first deliverable](#topic-list-page), but before that can happen, [prerequisites](#prerequisites) need to be discussed, agreed and implemented.
+
+#### Prerequisites
+
+In order to provide any UI, the whole stack which will support it will need to be discussed and considered. This will include (but not be limited to):
+
+- Setting up the required UI build/packaging for deployment to Strimzi
+- Integrating the UI with Strimzi itself
+- A Kafka backend server for Kafka data/requests the UI server can call
+- UI server configurability/capability - including aspects such as;
+  - Transport security
+  - User session, authentication and authorization capabilities
+  - Cluster metadata - listener addresses, number of brokers, etc
+  - Maintenance/tracing support
+- UI client side capabilities
+  - State design
+  - Configuration/capability discovery from the server
+  - Routing and navigation logic
+  - Maintenance/tracing capabilities
+
+I am more than happy to elaborate and collaborate on any of these points here, and would suggest these (and any follow items identified while discussing this proposal) are completed before any client side UI work begins.
+
+#### Topic list page
+
+Given the prerequisites are satisfied, the first user task I would suggest is implemented is a listing of all the topics in a given cluster. This will exercise all the prerequisites, as well as offer a user of Strimzi a new way to interact with their deployment. Using a [gherkin](https://cucumber.io/docs/gherkin/reference/) behavioural syntax to describe the capability, it would:
```

Comment:
is it limited just to listing the topic names or even showing basic information like number of partitions and replication factor?

Reply:
I would suggest in the first phase we only fetch and show data. Your correct, a name is a bit meaningless on its own! More than happy to expand this first deliverable with other items, such as partition count and replication factor (as shown in the mockup).